Eatery/Controllers/BRB/BRBAccountManager.swift

```diff
@@ -0,0 +1,296 @@
+                    block()
+                    return
+                }
+
+                if let sessionId = urlComponents.queryItems?.first(where: { $0.name == "sessionId" })?.value {
+                    self.stage = .finished(sessionId: sessionId)
+                } else {
+                    self.stage = .loginFailed
+                }
+            } else {
+                self.stage = .transition
+            }
+            
+            //run block for stage
+            block()
+        })
+    }
+}
+
+class BRBAccountManager {
+    
+    private var connectionHandler: BRBConnectionHandler!
+    var delegate: BRBAccountManagerDelegate?
+    
+    init() {
+        connectionHandler = BRBConnectionHandler()
+        connectionHandler.delegate = self
+    }
+    
+    func saveLoginInfo(loginInfo: LoginInfo) {
+        BRBAccountSettings.saveToKeychain(loginInfo: loginInfo)
+    }
+    
+    func removeSavedLoginInfo() {
+        BRBAccountSettings.removeKeychainLoginInfo()
+    }
+    
+    func getCredentials() -> LoginInfo? {
+        return BRBAccountSettings.loadFromKeychain()
+    }
+    
+    @objc func queryCachedBRBData() {
+        if let (netid, password) = BRBAccountSettings.loadFromKeychain() {
+            connectionHandler.netid = netid
+            connectionHandler.password = password
+            connectionHandler.handleLogin()
+        }
+    }
+    
+    func queryBRBData(netid: String, password: String) {
+        connectionHandler.netid = netid
+        connectionHandler.password = password
+        connectionHandler.handleLogin()
+    }
+    
+    func getConnectionStage() -> Stages {
+        return connectionHandler.stage
+    }
+    
+    func getCachedAccount() -> BRBAccount? {
+        if let brbAccount = UserDefaults.standard.object(forKey: "BRBAccount") as? Data {
+            let decoder = JSONDecoder()
+            if let loadedBRBAccount = try? decoder.decode(BRBAccount.self, from: brbAccount) {
+                return loadedBRBAccount
+            }
+        }
+        return nil
+    }
+}
+
+extension BRBAccountManager: BRBConnectionDelegate {
+    func retrievedSessionId(id: String) {
+        NetworkManager.shared.getBRBAccountInfo(sessionId: id) { [weak self] (account, error) in
+            guard let self = self else {
+                return
+            }
+            
+            if let account = account {
+                
+                let encoder = JSONEncoder()
+                if let encoded = try? encoder.encode(account) {
+                    let defaults = UserDefaults.standard
+                    defaults.set(encoded, forKey: "BRBAccount")
+                }
+                self.delegate?.queriedAccount(account: account)
+                
+            } else {
+                self.loginFailed(with: "Unable to parse account")
+            }
+        }
+    }
+    
+    func loginFailed(with error: String) {
+        self.delegate?.failedToGetAccount(with: error)
+    }
+    
```

Comment:
I think with eatery we usually have either zero or one newlines at the end of class definitions 

```
     func { 
     }
}
```

Reply:
Or 

```
    func {
    }

}
```